tools/version.sh

```diff
@@ -71,12 +71,12 @@ done
 OUTFILE=$1
 
 if [ -z ${VERSION} ] ; then
-  VERSION=`git -C ${WD} describe 2>/dev/null | tail -1 | cut -d'-' -f2`
+  VERSION=`git -C ${WD} describe --match "nuttx-*" 2>/dev/null | tail -1 | cut -d'-' -f2`
```

Comment:
This command currently returns **8.2** on my environment.
What's the real purpose for it?
Is it still useful or could we remove it in favor of retrieving the version from the `tag` command?

Reply:
It is probably necessary to get the right version on the release branches like as `releases/9.1`. 
The reason I added `--match` is just in case so that we can tag like `v1.2.3` without "nuttx-" locally.

Reply:
Ok I think I got it. It is tagged

`git fetch nuttx -f --tags`

Then run it.

Reply:
> This command currently returns 8.2 on my environment. 

That's because it's run from master.  8.2 is the last tag that's reachable from master. From 9.0 we started using release branches and tagging there instead from master.

> The reason I added --match is just in case so that we can tag like v1.2.3 without "nuttx-" locally.

@masayuki2009 changed the regex to accommodate something similar, but maybe this doesn't cover your use case.

Reply:
Adding `--match` option of `git describe` is nice-to-have to prevent the wrong version from being retrieved depending on the environment. If it's better without it or is side-effect, I'll delete it.

> @masayuki2009 changed the regex to accommodate something similar, but maybe this doesn't cover your use case.

This is about `git tag ... | grep -E "nuttx-` and this regex is helpful for me to manage tags on fork repository.

Thanks.

Reply:
> It is probably necessary to get the right version on the release branches like as `releases/9.1`.

If that is the reason, then using `tag` should be enough, since it makes no distinction of the current checked-out branch.
Right?

Reply:
Yes. My understanding is that the 1st check of `git describe` is mainly for release branches and 2nd check of `git tag` is for the master or other branches.